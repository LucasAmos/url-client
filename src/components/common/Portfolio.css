.portfolio {
  display: flex;
  align-items: center;
  background-color: rgba(226, 226, 226, 0.2);

  min-height: calc(100vh - 63px - 75px);
}

.portfolio input,
.portfolio button {
  font-family: 'Roboto';
  font-weight: 100;
}

.portfolio .view-work-col {
  text-align: center;
}

.portfolio h2 {
  font-family: 'Roboto Condensed';
  font-size: 40px;
}

.portfolio h5 {
  font-family: 'Roboto Condensed';
  font-size: 22px;
  font-weight: 600;
}

.portfolio .submit-button {
  width: 80px;
}

.portfolio .submit-button.btn.btn-outline-secondary {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.portfolio .submit-button.btn.btn-outline-secondary:hover {
  background-color: rgba(0, 0, 0, 0.1);
  color: #6c757d;
}

.portfolio .submit-button.btn.btn-outline-secondary:active {
  border-color: #ced4da;
}

.portfolio .btn-outline-secondary {
  border: 1px solid #ced4da;
}

@media (max-width: 500px) {
  .portfolio {
  }
}

@media (max-width: 430px) {
  .portfolio h2 {
    font-size: 36px;
  }
}

@media (max-width: 360px) {
  .portfolio h2 {
    font-size: 32px;
  }
}

@media (min-width: 400px) and (max-width: 575px) {
  .portfolio .input-group {
    padding-left: 5%;
    padding-right: 5%;
  }
}
