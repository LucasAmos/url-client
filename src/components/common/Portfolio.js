import React, { useState } from 'react';
import { Container, Row, Col, Form, Button, InputGroup, FormControl } from 'react-bootstrap';
import NavigationWrapper from './NavigationWrapper';
import Fade from 'react-reveal/Fade';

import PortfolioItem from './PortfolioItem';
import loadingImage from '../../res/images/loading.gif';
import './Portfolio.css';
import config from '../../config/config';

const Portfolio = props => {
  const [loading, setLoading] = useState(false);
  const [password, setPassword] = useState('');
  const [route] = useState(process.env.NODE_ENV === 'development' ? config.developmentURL : config.productionURL);
  const [work, setWork] = useState();
  const [authError, setAuthError] = useState(false);
  const [title, setTitle] = useState('View some of my work');

  function getWork() {
    setLoading(true);
    fetch(`${route}authorize?password=${password}`)
      .then(response => response.json())
      .then(res => {
        if (res.error) {
          setLoading(false);
          setAuthError(true);
        } else {
          setLoading(false);
          setTitle('Some projects I have worked on');
          setWork(res.data);
          setAuthError(false);
        }
      })
      .catch(() => {
        setLoading(false);
        setAuthError(true);
      });
  }
  return (
    <NavigationWrapper>
      <Fade>
        <div className="portfolio">
          <Container>
            <Row>
              <Col className="view-work-col" lg={{ span: 12 }}>
                <h2>{title}</h2>
                {work && <h5>(click on the images!)</h5>}
              </Col>
              <Col
                xs={{ span: 12 }}
                sm={{ span: 8, offset: 2 }}
                md={{ span: 6, offset: 3 }}
                lg={{ span: 4, offset: 4 }}
              >
                {!work && (
                  <Form.Group>
                    <InputGroup>
                      <FormControl
                        onKeyDown={e => (e.key === 'Enter' ? getWork() : null)}
                        type="password"
                        isInvalid={authError}
                        onChange={e => setPassword(e.target.value)}
                        value={password}
                        placeholder="Enter the password"
                      />
                      <InputGroup.Append>
                        <Button className="submit-button" variant="outline-secondary" onClick={getWork}>
                          {loading ? <img src={loadingImage} alt="loading" /> : 'Submit'}
                        </Button>
                      </InputGroup.Append>
                      <Form.Control.Feedback type="invalid">Please enter a valid password</Form.Control.Feedback>
                    </InputGroup>
                  </Form.Group>
                )}
              </Col>
            </Row>
          </Container>
          {work && work.map(elem => <PortfolioItem key={elem.title} portfolio={elem} />)}
        </div>
      </Fade>
    </NavigationWrapper>
  );
};

export default Portfolio;
