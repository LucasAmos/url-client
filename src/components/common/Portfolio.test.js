import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Portfolio from './Portfolio';

describe('Portfolio component', () => {
  let savedError;
  beforeEach(() => {
    savedError = console.error;
    console.error = jest.fn();
  });

  afterEach(() => {
    console.error = savedError;
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      shallow(
        <Portfolio />,
      ),
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('matches the snapshot', () => {
    const tree = renderer.create(
      <Portfolio />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
