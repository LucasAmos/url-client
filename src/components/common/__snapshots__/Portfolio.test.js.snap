// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches the snapshot 1`] = `
<div
  className="react-reveal portfolio"
  style={
    Object {
      "opacity": undefined,
    }
  }
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="view-work-col col-lg-12"
      >
        <h2>
          View some of my work
        </h2>
      </div>
      <div
        className="col-lg-4 col-md-6 col-sm-8 col-12 offset-lg-4 offset-md-3 offset-sm-2"
      >
        <div
          className="form-group"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter the password"
              type="password"
              value=""
            />
            <div
              className="input-group-append"
            >
              <button
                className="submit-button btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Submit
              </button>
            </div>
            <div
              className="invalid-feedback"
            >
              Please enter a valid password
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
